tests/components/saunum/test_climate.py

```diff
@@ -0,0 +1,412 @@
+async def test_climate_set_hvac_mode_heat(
+    hass: HomeAssistant,
+    mock_config_entry: MockConfigEntry,
+) -> None:
+    """Test setting HVAC mode to HEAT."""
+    entity_id = "climate.saunum_leil"
+
+    coordinator = mock_config_entry.runtime_data
+    coordinator.client.async_start_session = AsyncMock()
+    coordinator.async_request_refresh = AsyncMock()
+
+    # Turn on heating
+    await hass.services.async_call(
+        CLIMATE_DOMAIN,
+        SERVICE_SET_HVAC_MODE,
+        {ATTR_ENTITY_ID: entity_id, ATTR_HVAC_MODE: HVACMode.HEAT},
+        blocking=True,
+    )
+    await hass.async_block_till_done()
+
+    coordinator.client.async_start_session.assert_called_once()
+    coordinator.async_request_refresh.assert_called_once()
+
+
+@pytest.mark.usefixtures("init_integration")
+async def test_climate_set_hvac_mode_off(
+    hass: HomeAssistant,
+    mock_config_entry: MockConfigEntry,
+) -> None:
+    """Test setting HVAC mode to OFF."""
+    entity_id = "climate.saunum_leil"
+
+    coordinator = mock_config_entry.runtime_data
+    coordinator.client.async_stop_session = AsyncMock()
+    coordinator.async_request_refresh = AsyncMock()
+
+    # Turn off heating
+    await hass.services.async_call(
+        CLIMATE_DOMAIN,
+        SERVICE_SET_HVAC_MODE,
+        {ATTR_ENTITY_ID: entity_id, ATTR_HVAC_MODE: HVACMode.OFF},
+        blocking=True,
+    )
+    await hass.async_block_till_done()
+
+    coordinator.client.async_stop_session.assert_called_once()
+    coordinator.async_request_refresh.assert_called_once()
+
+
+@pytest.mark.usefixtures("init_integration")
+async def test_climate_set_temperature(
+    hass: HomeAssistant,
+    mock_config_entry: MockConfigEntry,
+) -> None:
+    """Test setting target temperature."""
+    entity_id = "climate.saunum_leil"
+
+    coordinator = mock_config_entry.runtime_data
+    coordinator.client.async_set_target_temperature = AsyncMock()
+    coordinator.async_request_refresh = AsyncMock()
+
+    # Set temperature to 85°C
+    await hass.services.async_call(
+        CLIMATE_DOMAIN,
+        SERVICE_SET_TEMPERATURE,
+        {ATTR_ENTITY_ID: entity_id, ATTR_TEMPERATURE: 85},
+        blocking=True,
+    )
+    await hass.async_block_till_done()
+
+    coordinator.client.async_set_target_temperature.assert_called_once_with(85)
+    coordinator.async_request_refresh.assert_called_once()
+
+
+@pytest.mark.usefixtures("init_integration")
+async def test_climate_device_info(
+    hass: HomeAssistant,
+    mock_config_entry: MockConfigEntry,
+    entity_registry: er.EntityRegistry,
+) -> None:
+    """Test climate device info."""
+    entity_id = "climate.saunum_leil"
+    entity_entry = entity_registry.async_get(entity_id)
+    assert entity_entry is not None
+
+    state = hass.states.get(entity_id)
+    assert state is not None
+
+    # Check friendly name (should be just the device name since _attr_name = None)
+    assert state.attributes.get("friendly_name") == "Saunum Leil"
```

Comment:
in theory the name would be tested with a snapshot test. You could also add a snapshot test for the device in test_init.py